### PR TITLE
TA for simultaneous delete with copy object

### PR DESCRIPTION
**Pull request by archanabpatil**
Signed-off-by: archanabpatil <66412855+archanabpatil@users.noreply.github.com>

# Problem Statement
-TA for tests [TEST-45516] [TEST-45518]  [TEST-45519] [TEST-45523][TEST-45524][TEST-45525]

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by archanabpatil** [comment]
Test execution logs attached in https://jts.seagate.com/browse/CORTX-33467

File: tests/s3/copy_object/test_copy_obj_simultaneous_del.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        """Test Parallel copy and put operation from same source bucket"""
+        LOGGER.info("STARTED: Test Delete source object when copy simple object is in progress")
+        LOGGER.info("Upload a large object (~1GB) srcobj to srcbuck")
+        self.create_put_parallel_copy_and_delete_object(("src-buck", "src-obj", "dest-buck",
+                                                        "dest-obj", "NoSuchKey", 1024),
+                                                        ("src-buck", "src-obj"))
+        LOGGER.info("Upload a large object (~5GB) srcobj to srcbuck")
+        self.create_put_parallel_copy_and_delete_object(("src-buck", "src-obj", "dest-buck",
+                                                         "dest-obj", "NoSuchKey", 5*1024),
+                                                        ("src-buck", "src-obj"))
```
Comment (ketanarlulkar) [inline tests/s3/copy_object/test_copy_obj_simultaneous_del.py:133]: Have a for loop for object size

File: tests/s3/copy_object/test_copy_obj_simultaneous_del.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        else:
+            self.create_put_object(copy_args[0], copy_args[1], copy_args[5])
+        LOGGER.info("Parallely, \n1. From %s copy %s to %s as %s\n2. Delete %s from %s",
+                    copy_args[0], copy_args[1], copy_args[2], copy_args[3], delete_args[1],
+                    delete_args[1])
+        with multiprocessing.Pool(processes=2) as pool:
+            process1 = pool.apply_async(self.copy_object_wrapper, args=copy_args[:5])
+            process2 = pool.apply_async(self.delete_object_wrapper, args=delete_args)
+            assert_utils.assert_true(process1.get()[0], process1.get()[1])
+            assert_utils.assert_true(process2.get()[0], process2.get()[1])
```
Comment (ketanarlulkar) [inline tests/s3/copy_object/test_copy_obj_simultaneous_del.py:117]: Need to add verify either NoSuchKey error on copy object or both succeed

Comment (archanabpatil) [reply tests/s3/copy_object/test_copy_obj_simultaneous_del.py]: This is being taken care by copy_object_wrapper method

**Review by apoorva-rao — APPROVED** [review]

**Comment by archanabpatil** [comment]
Added collection logs in jira ticket

**Review by ketanarlulkar — APPROVED** [review]

File: tests/s3/copy_object/test_copy_obj_simultaneous_del.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            if is_mpu:
+                resp = system_utils.create_file(fpath=copy_args[6], count=obj_size, b_size="1M")
+                assert_utils.assert_true(resp[0], resp[1])
+                resp = self.s3mp_obj.simple_multipart_upload(copy_args[0], copy_args[1], obj_size,
+                                                             copy_args[6], copy_args[5])
+            else:
+                self.create_put_object(copy_args[0], copy_args[1], obj_size)
+            LOGGER.info("Parallely, \n1. From %s copy %s to %s as %s\n2. Delete %s from %s",
+                        copy_args[0], copy_args[1], copy_args[2], copy_args[3], delete_args[1],
+                        delete_args[1])
```
Comment (kapiljinna) [inline tests/s3/copy_object/test_copy_obj_simultaneous_del.py:134]: Line 134 : It should be delete_args[0] , can you confirm this ?

Comment (archanabpatil) [reply tests/s3/copy_object/test_copy_obj_simultaneous_del.py:134]: yes, thanks

**Review by kapiljinna — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**